test/unit/test_doctor.py

```diff
@@ -0,0 +1,85 @@
+import os
+from collections.abc import Generator
+from contextlib import contextmanager
+from unittest.mock import patch
+
+import pytest
+
+from exasol_integration_test_docker_environment.doctor import (
+    SUPPORTED_PLATFORMS,
+    HealthProblem,
+    diagnose_docker_daemon_not_available,
+    is_docker_daemon_available,
+    is_supported_platform,
+)
+
+
+@pytest.fixture
+def temporary_env_factory():
+    """
+    Creates a temporary environment factory that returns a generator that yields a temporary environment.
+    """
+
+    @contextmanager
+    def temporary_env_factory(env_vars: dict) -> Generator[os._Environ, None, None]:
+        """
+        Creates a temporary environment, containing the current environment variables.
+
+        :param env_vars: to be updated within the temporary environment.
+
+        :return: the temporary environment variables in use
+        """
+        old_env = dict(os.environ)
+        os.environ.update(env_vars)
+        yield os.environ
+        os.environ.clear()
+        os.environ.update(old_env)
+
+    return temporary_env_factory
+
+
+def test_docker_connection_attempt_on_non_existing_unix_socket_returns_false(
+    temporary_env_factory,
```

Comment:
maybe this fixture can be replaced by `mock.patch.dict(os.environ, clear=True)` or `monkeypatch`:
https://stackoverflow.com/questions/77255758/how-can-i-mock-my-environment-variables-for-my-pytest

Reply:
Done